models/Asset.php

```diff
@@ -1963,6 +1964,34 @@ public function clearThumbnails($force = false)
         }
     }
 
+    /**
+     * @param string $oldPath
+     *
+     * @return array
+     *
+     * @throws \League\Flysystem\FilesystemException
+     */
+    private function updateChildPaths(string $oldPath)
```

Comment:
Just moving the parent folder should be enough, right? Because all children of the parent will be moved anyway when calling `$storage->move()` for it, or what is the reason for that? 
If not we'd have to think about a more robust solution, because this only works for a certain amount of children (I'm just thinking about a folder with millions of assets in it 😁).

Reply:
`$storage->move()` fails when you move a sub folder to another folder(S3 does not allow you to move directories), so this is required so that children of parent folder are also moved as well.